### PR TITLE
linuxPackages.drbd: 9.2.8 -> 9.2.9

### PR DESCRIPTION
**Pull request by birkb**
## Description of changes

- [6.8 kernel support](https://github.com/LINBIT/drbd/releases/tag/drbd-9.2.9) added
- aligned to [Nix standard build environment](https://ryantm.github.io/nixpkgs/stdenv/stdenv/)
- test added
- redundant `installPhase` and `postFixup` definitions removed

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/all-tests.nix [hunk 1]
```diff
@@ -266,6 +266,7 @@ in {
   dovecot = handleTest ./dovecot.nix {};
   drawterm = discoverTests (import ./drawterm.nix);
   drbd = handleTest ./drbd.nix {};
+  drbd-driver = handleTest ./drbd-driver.nix {};
```
Comment (Thesola10) [inline nixos/tests/all-tests.nix:269]: The `drbd` test above uses the `services.drbd.enable` option, which already enables the kernel module. This test is redundant.

https://github.com/NixOS/nixpkgs/blob/44072e24566c5bcc0b7aa9178a0104f4cfffab19/nixos/modules/services/network-filesystems/drbd.nix#L43

**Review by Thesola10 — CHANGES_REQUESTED** [review]
Correct me if I'm wrong, but the following commits do not fit NixOS contributor guidelines. All commits should be squashed into the first one, or split according to their semantic effect.

image

File: nixos/tests/drbd-driver.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  name = "drbd-driver";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ birkb ];
+  };
+
+  nodes = {
+    machine = { config, pkgs, ... }: {
+      boot.kernelModules = [ "drbd" ];
+      boot.extraModulePackages = with config.boot.kernelPackages; [ drbd ];
+      boot.kernelPackages = pkgs.linuxPackages;
```
Comment (astro) [inline nixos/tests/drbd-driver.nix:11]:
```suggestion
      boot = {
        kernelModules = [ "drbd" ];
        extraModulePackages = with config.boot.kernelPackages; [ drbd ];
        kernelPackages = pkgs.linuxPackages;
      };
```

File: nixos/tests/drbd-driver.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import ./make-test-python.nix ({ lib, pkgs, ... }: {
+  name = "drbd-driver";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ birkb ];
+  };
```
Comment (astro) [inline nixos/tests/drbd-driver.nix:5]:
```suggestion
  meta.maintainers = with pkgs.lib.maintainers; [ birkb ];
```

**Review by astro — CHANGES_REQUESTED** [review]

File: nixos/tests/all-tests.nix (hunk 1 above)
Comment (birkb) [reply nixos/tests/all-tests.nix:269]: The [drbd](https://github.com/birkb/nixpkgs/blob/drbd-kernel-module/nixos/tests/drbd.nix) test validates the userland utilities, but not the kernel driver. The in-tree driver is loaded to create the DRBD devices, but no special driver checks are done.

The [drbd-driver](https://github.com/birkb/nixpkgs/blob/drbd-kernel-module/nixos/tests/drbd-driver.nix) test loads the out-of-tree driver (provided by this package) and validates the version of the loaded driver.

I think we need both tests...

File: nixos/tests/drbd-driver.nix (hunk 2 above)
Comment (birkb) [reply nixos/tests/drbd-driver.nix]: Hi Astro, thanks for the hint. I have tweaked both areas.

**Comment by K900** [comment]
Mass ping, please resubmit.

**Pull request closed without merging**